### PR TITLE
Fixed - IDENTITY-4233

### PR DESCRIPTION
**Pull request by JKAUSHALYA**
Changed the AuthenticatedUser calss to use UserCoreUtill class to get username.
https://wso2.org/jira/browse/IDENTITY-4233

### CODE REVIEW

File: components/authentication-framework/org.wso2.carbon.identity.application.authentication.framework/src/main/java/org/wso2/carbon/identity/application/authentication/framework/model/AuthenticatedUser.java [hunk 1]
```diff
@@ -171,10 +171,10 @@ public void setUserAttributes(Map<ClaimMapping, String> userAttributes) {
     public String getUsernameAsSubjectIdentifier(boolean useUserstoreDomainInLocalSubjectIdentifier, boolean useTenantDomainInLocalSubjectIdentifier){
```
Comment (omindu) [inline components/authentication-framework/org.wso2.carbon.identity.application.authentication.framework/src/main/java/org/wso2/carbon/identity/application/authentication/framework/model/AuthenticatedUser.java]: Formatting issue.

File: components/authentication-framework/org.wso2.carbon.identity.application.authentication.framework/src/main/java/org/wso2/carbon/identity/application/authentication/framework/model/AuthenticatedUser.java [hunk 2]
```diff
@@ -171,10 +171,10 @@ public void setUserAttributes(Map<ClaimMapping, String> userAttributes) {
...
         String userName = this.userName;
         if (useUserstoreDomainInLocalSubjectIdentifier && userStoreDomain != null){
```
Comment (omindu) [inline components/authentication-framework/org.wso2.carbon.identity.application.authentication.framework/src/main/java/org/wso2/carbon/identity/application/authentication/framework/model/AuthenticatedUser.java]: Formatting issue.

File: components/authentication-framework/org.wso2.carbon.identity.application.authentication.framework/src/main/java/org/wso2/carbon/identity/application/authentication/framework/model/AuthenticatedUser.java (hunk 1 above)
Comment (JKAUSHALYA) [inline components/authentication-framework/org.wso2.carbon.identity.application.authentication.framework/src/main/java/org/wso2/carbon/identity/application/authentication/framework/model/AuthenticatedUser.java]: Fixed in https://github.com/wso2/carbon-identity/pull/1690

File: components/authentication-framework/org.wso2.carbon.identity.application.authentication.framework/src/main/java/org/wso2/carbon/identity/application/authentication/framework/model/AuthenticatedUser.java (hunk 2 above)
Comment (JKAUSHALYA) [inline components/authentication-framework/org.wso2.carbon.identity.application.authentication.framework/src/main/java/org/wso2/carbon/identity/application/authentication/framework/model/AuthenticatedUser.java]: Fixed in https://github.com/wso2/carbon-identity/pull/1690

File: components/application-authenticators/org.wso2.carbon.identity.application.authenticator.social/src/main/java/org/wso2/carbon/identity/application/authenticator/social/live/WindowsLiveOAuth2Authenticator.java [hunk 3]
```diff
@@ -131,6 +131,15 @@ protected String getScope(String scope, Map<String, String> authenticatorPropert
     }
 
     /**
+     * We do not have a default claim dialect.
```
Comment (amilam) [inline components/application-authenticators/org.wso2.carbon.identity.application.authenticator.social/src/main/java/org/wso2/carbon/identity/application/authenticator/social/live/WindowsLiveOAuth2Authenticator.java]: This comment is not good enough. What does "We do not have a default claim dialect" mean as a method comment. A method level should explain what that method does. Also, you should mention what is returned by the method.

File: components/application-authenticators/org.wso2.carbon.identity.application.authenticator.social/src/main/java/org/wso2/carbon/identity/application/authenticator/social/yahoo/YahooOAuth2Authenticator.java [hunk 4]
```diff
@@ -121,6 +121,15 @@ protected String getTokenEndpoint(Map<String, String> authenticatorProperties) {
     }
 
     /**
+     * We do not have a default claim dialect.
+     * @return
+     */
```
Comment (amilam) [inline components/application-authenticators/org.wso2.carbon.identity.application.authenticator.social/src/main/java/org/wso2/carbon/identity/application/authenticator/social/yahoo/YahooOAuth2Authenticator.java]: Please improve this comment as advised in my previous comment.

File: components/application-authenticators/org.wso2.carbon.identity.application.authenticator.social/src/main/java/org/wso2/carbon/identity/application/authenticator/social/live/WindowsLiveOAuth2Authenticator.java (hunk 3 above)
Comment (JKAUSHALYA) [inline components/application-authenticators/org.wso2.carbon.identity.application.authenticator.social/src/main/java/org/wso2/carbon/identity/application/authenticator/social/live/WindowsLiveOAuth2Authenticator.java]: Fixed in https://github.com/wso2/carbon-identity/pull/1701

File: components/application-authenticators/org.wso2.carbon.identity.application.authenticator.social/src/main/java/org/wso2/carbon/identity/application/authenticator/social/yahoo/YahooOAuth2Authenticator.java (hunk 4 above)
Comment (JKAUSHALYA) [inline components/application-authenticators/org.wso2.carbon.identity.application.authenticator.social/src/main/java/org/wso2/carbon/identity/application/authenticator/social/yahoo/YahooOAuth2Authenticator.java]: Fixed in https://github.com/wso2/carbon-identity/pull/1701

**Pull request merged**